### PR TITLE
Added sampling tests

### PR DESCRIPTION
**Pull request by Lyrcaxis**
Added some tests for testing various levels of the sampling pipeline -- from low-level logits to high-level `SafeLLamaSamplerChainHandle`. Includes both unbatched and batched sampling.

### CODE REVIEW

File: LLama.Unittest/SamplingTests.cs [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        private readonly ITestOutputHelper _testOutputHelper;
+        private readonly LLamaWeights _model;
+        private readonly ModelParams _params;
+
+        private readonly LLamaBatch _batch;
+        private readonly StreamingTokenDecoder _decoder;
+
+        public void Dispose() => _model.Dispose();
+
+        private unsafe Span<float> GetLogits(LLamaContext context, int totalSequences) => new(llama_get_logits(context.NativeHandle), totalSequences * _model.VocabCount);
```
Comment (martindevans) [inline LLama.Unittest/SamplingTests.cs:23]: This is already available in the SafeLLamaContextHandle with a safe wrapper. See [here](https://github.com/SciSharp/LLamaSharp/blob/4436f7e28fcac6e792e6af364f899729d68ba778/LLama/Native/SafeLLamaContextHandle.cs#L441-L469).

**Review by martindevans — CHANGES_REQUESTED** [review]
Just one changed requested :)

File: LLama.Unittest/SamplingTests.cs (hunk 1 above)
Comment (martindevans) [reply LLama.Unittest/SamplingTests.cs:23]: [Discussed in Discord](https://discord.com/channels/1106946823282761851/1106947264938790972/1306409468547956798).

tl;dr: that method isn't quite suitable. Lyrcaxis will modify it.

Comment (Lyrcaxis) [reply LLama.Unittest/SamplingTests.cs]: Added the option to retrieve multiple logits of multiple tokens and documentation for it on the existing method:
image

lmk if you feel documentation is lacking or excessive.
